packages/openapi-parser/src/utils/bundle.ts

```diff
@@ -0,0 +1,253 @@
+import fs from 'node:fs/promises'
+import path from 'node:path'
+import type { UnknownObject } from '../types'
+import { getSegmentsFromPath } from './getSegmentsFromPath'
+import { isObject } from './isObject'
+import { normalize } from './normalize'
+import { dereference } from './dereference'
+
+/**
+ * Checks if a string is a remote URL (starts with http:// or https://)
+ * @param value - The URL string to check
+ * @returns true if the string is a remote URL, false otherwise
+ * @example
+ * ```ts
+ * isRemoteUrl('https://example.com/schema.json') // true
+ * isRemoteUrl('http://api.example.com/schemas/user.json') // true
+ * isRemoteUrl('#/components/schemas/User') // false
+ * isRemoteUrl('./local-schema.json') // false
+ * ```
+ */
+export function isRemoteUrl(value: string): boolean {
+  return value.startsWith('http://') || value.startsWith('https://')
+}
+
+/**
+ * Checks if a string is a local reference (starts with #)
+ * @param value - The reference string to check
+ * @returns true if the string is a local reference, false otherwise
+ * @example
+ * ```ts
+ * isLocalRef('#/components/schemas/User') // true
+ * isLocalRef('https://example.com/schema.json') // false
+ * isLocalRef('./local-schema.json') // false
+ * ```
+ */
+export function isLocalRef(value: string): boolean {
+  return value.startsWith('#')
+}
+
+type ResolveResult = { ok: true; data: unknown } | { ok: false }
+
+/**
+ * Fetches and normalizes data from a remote URL
+ * @param url - The URL to fetch data from
+ * @returns A promise that resolves to either the normalized data or an error result
+ * @example
+ * ```ts
+ * const result = await fetchUrl('https://api.example.com/data.json')
+ * if (result.ok) {
+ *   console.log(result.data) // The normalized data
+ * } else {
+ *   console.log('Failed to fetch data')
+ * }
+ * ```
+ */
+export async function fetchUrl(url: string): Promise<ResolveResult> {
+  try {
+    const result = await fetch(url)
+
+    if (result.ok) {
+      const body = await result.text()
+
+      return {
+        ok: true,
+        data: normalize(body),
+      }
+    }
+
+    return {
+      ok: false,
+    }
+  } catch {
+    return {
+      ok: false,
+    }
+  }
```

Comment:
Promise chain is cleaner here. 
```suggestion
    return await fetch(url).then(res => res.ok ? {
      ok: true,
      data: normalize(await res.text()),
    } : {
      ok: false,
    }
  ).catch(() => ({
    ok: false,
  }))
```